interview-questions.md

```diff
@@ -3,44 +3,48 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
+  Your answer: animal_id, the foreign key would go to sightings because an animal can have many sightings at different times.
 
   Researched answer:
 
 
 
 2. Which routes must always be passed params and why?
 
-  Your answer:
+  Your answer: params.require
 
   Researched answer:
 
 
 
 3. Write a rails route for a controller called "main" and a page called "game" that takes in a parameter called "guess".
 
-  Your answer:
+  Your answer: get '/main/game, to: 'main#guess'
 
 
 
 4. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: rails g resource does all of them for you so it creates the model, controller, then just run db:create
```

Comment:
rails g migration - creates a migration to change the database 
rails g model - create a new table with all the table attributes 
rails g controller - create a new controller for a specific model 
rails g resource - create the routes, model, controller